### PR TITLE
Add missing client routes

### LINKED ISSUE #75: [Client] Create routes to create an event/alert/media linked to a fire.
From Mateo's request:

> En gros j'aimerais qu'on mette en place une petite présentation ou je filme un départ de feu et je ping l'api quand j'en détecte un et du coup ca vous permet d'enchaîner sur la présentation de l'api. Je dois utiliser quoi pour ca ?

It seems we lack the proper client's routes (and potentially API routes) to achieve that.

### PR DESCRIPTION
**Pull request by florianriche**
Add missing client routes mentionned in #75 .
Tested on the local API.

closes #75 

NB: the media is not yet uploaded as we don't have included the Qarnot computing bucket service.

### CODE REVIEW

File: client/pyroclient/exceptions.py [hunk 1]
```diff
@@ -7,3 +7,13 @@ def __init__(self, status_code: int, response_message: str = None) -> None:
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
         return f"{class_name}(status_code={self.status_code!r}, response_message={self.response_message!r})"
+
+
+class MissingArgumentException(Exception):
```
Comment (frgfm) [inline client/pyroclient/exceptions.py:12]: wasn't a ValueError enough (with a proper message I mean)

File: client/pyroclient/client.py [hunk 2]
```diff
@@ -61,21 +64,69 @@ def hearbeat(self):
...
         """Create a media entry"""
-        return requests.post(self.routes["create-media"], headers=self.headers)
+        return requests.post(self.routes["create-media"], headers=self.headers, json={"device_id": device_id})
+
+    def create_media_from_device(self):
+        """Create a media entry from a device (no need to specify device ID)"""
+        return requests.post(self.routes["create-media-from-device"], headers=self.headers, json={})
 
-    def upload_media(self):
+    def upload_media(self, media_id, image_data):
```
Comment (frgfm) [inline client/pyroclient/client.py:126]: some typing over here would be nice!

**Review by frgfm — COMMENTED** [review]
Thanks for the PR! Just a few comments but nothing major!

File: client/pyroclient/client.py (hunk 2 above)
Comment (florianriche) [reply client/pyroclient/client.py:126]: Indeed, done

File: client/pyroclient/exceptions.py (hunk 1 above)
Comment (florianriche) [reply client/pyroclient/exceptions.py:12]: I wanted to make it more explicit but ok, let's switch to ValueError.

**Comment by frgfm** [comment]
For the sake of adding unittests in the same PR, perhaps let's first merge #74 before this one

**Comment by florianriche** [comment]
The client methods I added are "post" routes. Did you find a way to unit test the post routes?

File: client/README.md [hunk 3]
```diff
@@ -55,14 +56,25 @@ api_client = client.Client(API_URL, CREDENTIALS_LOGIN, CREDENTIALS_PASSWORD)
...
 Use it to query alerts:
 ```python
-# Send alerts:
-api_client.send_alert()
 
-# Send medias:
-api_client.send_medias()
+#AS A DEVICE:
+## Create a device
+event_id = client.create_event(lat=10, lon=10).json()["id"]
```
Comment (fe51) [inline client/README.md:62]: Should not it be the instance of the class Client, api_client, to use in the following examples ?
```suggestion
event_id = api_client.create_event(lat=10, lon=10).json()["id"]
```

File: client/README.md [hunk 4]
```diff
@@ -55,14 +56,25 @@ api_client = client.Client(API_URL, CREDENTIALS_LOGIN, CREDENTIALS_PASSWORD)
...
+## Create a media
+media_id = client.create_media_from_device().json()["id"]
```
Comment (fe51) [inline client/README.md:64]: same here

File: client/README.md [hunk 5]
```diff
@@ -55,14 +56,25 @@ api_client = client.Client(API_URL, CREDENTIALS_LOGIN, CREDENTIALS_PASSWORD)
...
+## Create an alert linked to the media and the event
+client.send_alert_from_device(lat=10, lon=10, event_id=event_id, media_id=media_id)
```
Comment (fe51) [inline client/README.md:66]: same here

File: client/README.md [hunk 6]
```diff
@@ -55,14 +56,25 @@ api_client = client.Client(API_URL, CREDENTIALS_LOGIN, CREDENTIALS_PASSWORD)
...
+
+## Upload an image on the media
+dummy_image = "https://ec.europa.eu/jrc/sites/jrcsh/files/styles/normal-responsive/" \
+                + "public/growing-risk-future-wildfires_adobestock_199370851.jpeg"
+image_data = requests.get(dummy_image)
+client.upload_media(media_id=media_id, image_data=image_data.content)
```
Comment (fe51) [inline client/README.md:72]: same here

**Comment by fe51** [comment]
Thanks for the PR ! I am trying to test it locally, and I have so far some issues (but it could came from my understanding :)) 

Have tested all the new methods  ?

Also, I have jsute noticed, the heartbeat method is actually called "hearbeat" , I think this is just typo but we can correct it in the PR :)

**Comment by florianriche** [comment]
Yes I tested it locally; the client need to be logged as a device though.

**Comment by frgfm** [comment]
> The client methods I added are "post" routes. Did you find a way to unit test the post routes?

Good point, so far we haven't solved that, so I guess we can do that later on!

**Review by frgfm — APPROVED** [review]
Thanks for the edits!

**Pull request merged**